### ISSUE TITLE
JDL fails to compile: bad type in conditional expression

### ISSUE THREAD

**Issue by mraible** [jhipster/generator-jhipster#14328]
##### **Overview of the issue**

I'm trying to re-create the backend I used in my [Mobile JHipster](https://developer.okta.com/blog/2020/04/27/mobile-development-ionic-react-native-jhipster) presentation. The app I'm trying to create is a bit like my 21-Points Health app, but I changed it to use OAuth, React, and WebFlux.

```
/*
 * 21-Points (https://github.com/mraible/21-points) as a WebFlux + React app
 */

application {
  config {
    applicationType monolith
    reactive true
    baseName HealthPoints
    packageName com.okta.developer
    authenticationType oauth2
    prodDatabaseType postgresql
    buildTool gradle
    testFrameworks [cypress]
    clientFramework react
    enableTranslation true
    nativeLanguage en
    languages [en, es]
  }
  entities Points, BloodPressure, Weight, Preferences
}

// JDL definition for application 'TwentyOnePoints' generated with command 'jhipster export-jdl'

entity BloodPressure {
  timestamp ZonedDateTime required,
  systolic Integer required,
  diastolic Integer required
}
entity Weight {
  timestamp ZonedDateTime required,
  weight Double required
}
entity Points {
  date LocalDate required,
  exercise Integer,
  meals Integer,
  alcohol Integer,
  notes String maxlength(140)
}
entity Preferences {
  weeklyGoal Integer required min(10) max(21),
  weightUnits Units required
}

enum Units {
  KG,
  LB
}

relationship OneToOne {
  Preferences{user(login)} to User
}
relationship ManyToOne {
  BloodPressure{user(login)} to User,
  Weight{user(login)} to User,
  Points{user(login)} to User
}

paginate BloodPressure, Weight with infinite-scroll
paginate Points with pagination
```

Running `jhipster jdl app.jdl` works fine, but when I try to compile it, I get an error:

```
> Task :compileJava
Note: Hibernate JPA 2 Static-Metamodel Generator 5.4.28.Final
/Users/mraible/dev/mobile-jhipster/backend/src/main/java/com/okta/developer/domain/Preferences.java:83: error: incompatible types: bad type in conditional expression
        this.userId = user != null ? user.getId() : null;
                                               ^
    String cannot be converted to Long
/Users/mraible/dev/mobile-jhipster/backend/src/main/java/com/okta/developer/domain/Preferences.java:89: error: incompatible types: bad type in conditional expression
        this.userId = user != null ? user.getId() : null;
                                               ^
    String cannot be converted to Long
/Users/mraible/dev/mobile-jhipster/backend/src/main/java/com/okta/developer/domain/Preferences.java:93: error: incompatible types: Long cannot be converted to String
        return this.userId;
                   ^
/Users/mraible/dev/mobile-jhipster/backend/src/main/java/com/okta/developer/domain/Preferences.java:97: error: incompatible types: String cannot be converted to Long
        this.userId = user;
                      ^
Note: /Users/mraible/dev/mobile-jhipster/backend/src/main/java/com/okta/developer/service/ColumnConverter.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
4 errors

> Task :compileJava FAILED

FAILURE: Build failed with an exception.
```

##### **Motivation for or Use Case**

It would be nice if JDLs I previously used still worked.

##### **Reproduce the error**

Copy the JDL above and run `jhipster jdl` to import it. Then run `./gradlew`.

##### **JHipster Version(s)**

`main` branch as of March 15, 2021, 18:00 MDT

**Comment by mraible** [comment]
If I modify `Preferences.java` to use `private String userId;`, it solves the problem. Seems like a problem with the entity generator, OAuth, and relationships to `User`.

**Comment by mraible** [comment]
Since I ran Sonar and Lighthouse on this app. I figured I'd share some results.

Sonar:

<img width="1436" alt="Screen Shot 2021-03-15 at 6 08 06 PM" src="https://user-images.githubusercontent.com/17892/111236987-b827eb80-85b9-11eb-9114-2ed4fb5fb5b4.png">

It's an improvement over [v6.8.0](https://speakerdeck.com/mraible/mobile-development-with-ionic-react-native-and-jhipster-allthetalks-2020?slide=26) and the 6.8.0 screenshot is not reactive!

Lighthouse:

<img width="1030" alt="Screen Shot 2021-03-15 at 6 16 04 PM" src="https://user-images.githubusercontent.com/17892/111237443-b7438980-85ba-11eb-9209-c2445c680828.png">

This is worse than [v6.8.0](https://speakerdeck.com/mraible/mobile-development-with-ionic-react-native-and-jhipster-allthetalks-2020?slide=27). It was all green back then.

I did uncomment the service worker in `index.html`.
